packages/backEnd/src/controller/advert.controller.ts

```diff
@@ -0,0 +1,31 @@
+import { Controller, Get, Post } from "@overnightjs/core";
+import { Request, Response, NextFunction } from "express";
+import { AddService } from "../services/adds.services";
+import { ICategories } from "../dto/user.dto";
+import { MulterRequest } from "../interface";
+
+@Controller("api/v1")
+export class AdvertController {
+  constructor(private addService: AddService = new AddService()) {}
+
+  @Get("filter")
+  async filterCategories(
+    req: Request<{}, {}, {}, ICategories>,
+    res: Response,
+    next: NextFunction
+  ) {
+    const response = await this.addService.filterCategories(req.query);
+    res.status(response.statusCode).json(response);
+  }
+  @Post("advertisement")
+  async addAdvertising(req: MulterRequest, res: Response, next: NextFunction) {
+    const response = await this.addService.addAdvertising(req.files);
+    res.status(response.statusCode).json(response);
+  }
+
+  @Get("data")
+  async advertisingData(req: Request, res: Response, next: NextFunction) {
```

Comment:
In POST we have addSomething. I would change the name to getAdvertisingData to be 100% sure that the method is used to get the data.

Reply:
Good point.